### PR TITLE
Featers/launcher page

### PR DESCRIPTION
**Pull request by cDarkDirk**
LauncherPage feature

### CODE REVIEW

File: src/api.js [hunk 1]
```diff
@@ -18,6 +18,52 @@ import {
...
+  // Host: localhost:8080
+  // Content-Length: 11
+  // test_id: 14
+
+  // template: TSMMortgage
+  // marker: TSMMortgageDebug
+  // start_time: 2018.02.05
+  //
+  // body - data
+  // http://localhost:8080/chains
```
Comment (Orumant) [inline src/api.js]: remove comments

File: src/api.js [hunk 2]
```diff
@@ -18,6 +18,52 @@ import {
...
+
+    const dataToSendLauncherPage = {
+      formName: formName,
+      template: templates,
+      paramData: formTemplate,
+      start_time: sheduleList
+   };
+
+  console.log(formName, formTemplate, sheduleList, templates);
```
Comment (Orumant) [inline src/api.js]: remove loging

File: src/components/Form/index.js [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+import "./style.css"
+import Calendar from "react-calendar";
+import DatePicker from "react-datepicker"
+import {submitFormTemplate} from "../../api";
+
+class MyForm extends React.Component {
+
+  componentDidMount() {
+    this.props.fetchBuilderChains()
+    // this.props.fetchFormTemplate(this.props.formName)
```
Comment (Orumant) [inline src/components/Form/index.js]: remove this

File: src/components/LauncherPage/index.js [hunk 4]
```diff
@@ -1,35 +1,42 @@
...
-                    <div> Time</div>
-                </Col>
-            </Row>
+  componentDidMount() {
+    this.props.fetchBuilderChains();
+  }
+
+  render() {
+    const {formBuilderChains, selectChainForm, launcher: {selectedForm}} = this.props;
+    //const form = formBuilderChains[launcher.selectedForm];
```
Comment (Orumant) [inline src/components/LauncherPage/index.js]: remove comment

File: src/components/LauncherPage/index.js [hunk 5]
```diff
@@ -1,35 +1,42 @@
...
+    return (
+      <div>
+        {/*<DropDownSelector id onSelect dropOptions selectedValue/>*/}
```
Comment (Orumant) [inline src/components/LauncherPage/index.js]: this one too

File: src/components/LauncherPage/index.js [hunk 6]
```diff
@@ -1,35 +1,42 @@
...
+                          dropOptions={formBuilderChains.map((option, index) => {
+                            return {value: option.name, label: option.name}
+                          })}
+                          selectedIndex={selectedForm}
+        />
+        {formBuilderChains.length &&
+          <div className="container">
             <Row>
-                <h2>FORM</h2>
+              <ScheduleForm></ScheduleForm>
```
Comment (Orumant) [inline src/components/LauncherPage/index.js:29]: replace it with self closing tag

File: src/components/LauncherPage/index.js [hunk 7]
```diff
@@ -1,35 +1,42 @@
...
             </Row>
-
-            <Row></Row>
-
-
-
-        </div>
-
-
+            <TemplateForm></TemplateForm>
```
Comment (Orumant) [inline src/components/LauncherPage/index.js:31]: replace it with self closing tag

File: src/components/LauncherPage/index.js [hunk 8]
```diff
@@ -1,35 +1,42 @@
...
+            <Row><Form formName={selectedForm}></Form></Row>
```
Comment (Orumant) [inline src/components/LauncherPage/index.js:32]: replace it with self closing tag

File: src/components/ScheduleForm/index.js [hunk 9]
```diff
@@ -1,60 +1,66 @@
...
-            </h2>
             <Row>
-                <Col>
-                    <div>Amount of times</div>
+              <FormGroup controlId="formHorizontalInput">
+                <Col componentClass={ControlLabel} sm={2}>
+                  <div>Amount of times: </div>
                 </Col>
-                <Col>
-                    <input></input>
```
Comment (Orumant) [inline src/components/ScheduleForm/index.js:21]: replace it with self closing tag

File: src/components/ScheduleForm/index.js [hunk 10]
```diff
@@ -1,60 +1,66 @@
...
                 </Col>
-                <Col>
-                    <div>Date</div>
+
+
+                <Col componentClass={ControlLabel} sm={2}>
+                  Date
                 </Col>
-                <Col>
-                    <DatePicker selected={props.date} onChange={props.changeDate}></DatePicker>
```
Comment (Orumant) [inline src/components/ScheduleForm/index.js:27]: replace it with self closing tag

File: src/components/ScheduleForm/index.js [hunk 11]
```diff
@@ -1,60 +1,66 @@
...
+
+                <Col sm={3}>
+                  <div>
+                    <DatePicker onChange={props.changeDate}
```
Comment (Orumant) [inline src/components/ScheduleForm/index.js:31]: replace it with self closing tag

File: src/components/ScheduleForm/index.js [hunk 12]
```diff
@@ -1,60 +1,66 @@
...
+
+
+                <Col componentClass={ControlLabel} sm={2}>
+                  Time
                 </Col>
-                <Col>
+
+                <Col sm={2}>
+                  <div>
                     <TimePicker value={props.time} onChange={props.changeTime}></TimePicker>
```
Comment (Orumant) [inline src/components/ScheduleForm/index.js:43]: replace it with self closing tag EVERYWHERE!

File: src/components/ScheduleForm/index.js [hunk 13]
```diff
@@ -1,60 +1,66 @@
...
-                    <option>Weekly</option>
-                </select></Col>
-            </Row>
 
-            {/*<Row>*/}
-                {/*<FormGroup controlId="InputForm">*/}
                 {/*<Col componentClass={ControlLabel} sm={2}>*/}
-                    {/*{field.label}*/}
+                  {/*Periodic:*/}
                 {/*</Col>*/}
```
Comment (Orumant) [inline src/components/ScheduleForm/index.js]: get rid of this huge comment

File: src/components/TemplateForm/index.js [hunk 14]
```diff
@@ -1,36 +1,58 @@
...
+
+
+  return (
+    <Jumbotron>
+      <FormGroup controlId="formHorizontalDropDown">
+        <Col componentClass={ControlLabel} sm={2}>
+          Template:
+        </Col>
+        <Col sm={10}>
+          {/*<FormControl componentClass="select"*/}
```
Comment (Orumant) [inline src/components/TemplateForm/index.js]: remove comment

File: src/containers/Form.js [hunk 15]
```diff
@@ -0,0 +1,26 @@
...
+        choosenDataTemplates: state.dataTemplate.choosenDataTemplates,
+        formBuilderChains: state.formBuilder.formBuilderChains || [],
+    }
+}
+
+function mapDispatchToProps(dispatch) {
+    return {
+        onFormInputChange: (value, paramName, formName) => dispatch(onFormInputChange(value, paramName, formName)),
+        fetchBuilderChains: () => dispatch(fetchBuilderChains()),
+  //    fetchFormTemplate: (formName) => dispatch(fetchFormTemplate(formName)),
```
Comment (Orumant) [inline src/containers/Form.js]: remove comments

**Review by Orumant — CHANGES_REQUESTED** [review]
Итог: Убирай закоментированные блоки кода, заменяй тэги на самозакрывающиеся

File: src/components/ScheduleForm/index.js (hunk 12 above)
Comment (cDarkDirk) [reply src/components/ScheduleForm/index.js:43]: Its not the html col, its bottstrap Col. It's not a self-closing component.

**Review by Orumant — APPROVED** [review]

**Pull request merged**